File: digiverifier/src/main/java/com/aashdit/digiverifier/config/superadmin/service/SuperAdminDashboardService.java
package com.aashdit.digiverifier.config.superadmin.service;

import com.aashdit.digiverifier.common.model.ServiceOutcome;
import com.aashdit.digiverifier.config.superadmin.dto.DashboardDto;
import com.aashdit.digiverifier.config.superadmin.dto.SuperAdminDashboardDto;

public interface SuperAdminDashboardService {

	ServiceOutcome<SuperAdminDashboardDto> getPendingDetails(SuperAdminDashboardDto superAdminDashboardDto);

	ServiceOutcome<SuperAdminDashboardDto> getActivityDetails(SuperAdminDashboardDto superAdminDashboardDto);

	ServiceOutcome<SuperAdminDashboardDto>  getUtilizationRatePerItem(SuperAdminDashboardDto superAdminDashboardDto);

	ServiceOutcome<SuperAdminDashboardDto>  getUtilizationRatePerReport(SuperAdminDashboardDto superAdminDashboardDto);

	ServiceOutcome<DashboardDto> getCompanyCountByActivity(DashboardDto dashboardDto);

}

File: digiverifier/src/app/services/superadmin-dashboard.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';

@Injectable({
  providedIn: 'root'
})
export class SuperadminDashboardService {

  constructor( private http:HttpClient) { }

  getTestData(){
    return this.http.get("https://jsonplaceholder.typicode.com/posts/1/comments");
  }
  getActivityDetails(data:any){
    return this.http.post(`${environment.apiUrl}/api/organization/getActivityDetails`, data);
  }

  getPendingDetails(data:any){
    return this.http.post(`${environment.apiUrl}/api/organization/getPendingDetails`, data);
  }

  getUtilizationRatePerItem(data:any){
    return this.http.post(`${environment.apiUrl}/api/organization/getUtilizationRatePerItem`, data);
  }

  getUtilizationRatePerReport(data:any){
    return this.http.post(`${environment.apiUrl}/api/organization/getUtilizationRatePerReport`, data);
  }

  getCompanyCountByActivity(data:any){
    return this.http.post(`${environment.apiUrl}/api/organization/getCompanyCountByActivity`, data);
  }

  postCompanyCountByActivity(data:any){
    return this.http.post(`${environment.apiUrl}/api/organization/getCompanyCountByActivity`, data);
  }

}
